sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -3002,14 +3002,17 @@ object SubtractDates {
   usage = "_FUNC_(sourceTz, targetTz, sourceTs) - Converts the timestamp without time zone `sourceTs` from the `sourceTz` time zone to `targetTz`. ",
   arguments = """
     Arguments:
-      * sourceTz - the time zone for the input timestamp
+      * sourceTz - the time zone for the input timestamp.
+                   If it is missed, the current session time zone is used as the source time zone.
```

Comment:
We don't have by-name parameters in SQL functions yet, and it's weird that the non-last parameter can be optional...

Reply:
Could you explain why do you think it is weird, and what kind of issues do you see.

Reply:
The main goal of the changes is to help users in migrations from other systems to Spark SQL easier. Other systems support optional first parameter:
- https://docs.aws.amazon.com/redshift/latest/dg/CONVERT_TIMEZONE.html
- https://docs.snowflake.com/en/sql-reference/functions/convert_timezone.html